client/components/submitProgram.js

```diff
@@ -52,13 +52,31 @@ async function _onClick(event) {
   dispatchEvent('submitProgram:output', output);
 }
 
+let MacCMDPressed = false;
+
+const MacCommandKeys = ['MetaLeft', 'MetaRight'];
+
+function _onKeyDown(event) {
+  if ((event.ctrlKey || MacCMDPressed) && event.keyCode === 13) {
+    _onClick(event);
+  }
+
+  if (MacCommandKeys.includes(event.code)) {
+    MacCMDPressed = true;
+  } else {
+    MacCMDPressed = false;
+  }
+}
+
 //init
 (() => {
   //DOM binding
   submitProgramElem = document.getElementById('submitBtn');
+  editorElement = document.getElementById('editor');
```

Comment:
@EricTurf in this project I'm using pub-sub design pattern to enable proper Separation of Concerns. Each component is responsible to deal wit their own Dom. submitProgram component shouldn't querySelect the codeEditor component. 

The components interact with each other by dispatching events and subscribing to events.

What you have to do is in the codeEditor component, add the _keyDown event listener and the whole `CMD + P` logic. When you detect that it was indeed `CMD + P` you dispatch an event say `codeEditor:runcode`. 
Then in the submitCode component, listen to the `codeEditor:runcode` event using `addEvent` helper. When the `codeEditor:runcode` is triggered, you have to call the _onClick method.

Reply:
For reference see how submitProgram and langSel components interact with each other.